### DISCUSSION TITLE
Easier Access - domestic heating

### DISCUSSION THREAD

**Discussion by arc12** [oemof/tespy#334] [Ideas]
I came across tespy when looking for approaches to simulating domestic central heating, specifically the flow rates, heat transfer rates to room, pressure drop in a multi-zone piping network with different diameter pipe-work with a standard domestic source (gas/oil boiler, ASHP, solar thermal, etc).
It looks like this would be possible, but because tespy does so much more, it is quite hard for a new-comer who is not an expert in thermal modelling to get started. For example, what components should be used to model a domestic "radiator" or fan coil, what are Q and pr for the HeatExchangerSimple, how would I model a TRV, or a 20% propylene glycol/water mixture..?? The barrier for getting started is just too high!

A tutorial covering this scenario would be most welcome. I cannot be the only person interested in modelling and controlling domestic heating in python (quite active open source energy monitoring and home automation communities exist).

**Comment by fwitte** [comment]
@arc12, thank you for reaching out and getting into contact, appreciate that!

For domestic central heating the software is indeed overengineered. However, I do think one could make this work, but it might take substantial effort. On the back-end, especially the solver (could be quite complex) and the components (introducing or adjusting components or specific parameters is much less work) things would need to be modified. Low flow rates (compared to power plant engineering) and non-active parts in the network, might be a challenge. Since I do not know too much about domestic heating systems, I have some questions myself:

- What is the overall goal of your simulation? Laying out the pipe-work and the radiators? Monitoring or optimization of existing systems, i.e. some kind of system control?
- The solver in TESPy solves for stationary operation, is that viable for domestic systems? Or are dynamic effects predominant? For example, consider a radiator, where the valve opens for a specific amount of time and then closes. After closing, the mass flow is zero, but the radiator still provides heat. Coupling TESPy for stationary operation with some kind of dynamic model could be interesting.

To answer your questions:

- The HeatExchangerSimple adds or extracts heat from a fluid. Q is the constant amount of heat exchanged with the environment and pr a constant outlet to inlet pressure ratio.
- If the heat exchanged with the environment should be the result of the type, size, ... of the radiator, then additional functionalities must be implemented on the component. This would however be a quite simple adjustment in TESPy.
- A TRV would regulate mass flow through the radiator to match a specific temperature in the room. Since TESPy calculates stationary operation, one could use an external method, that just sets the mass flow value through the radiator based on the temperature in the "ambient" (I imagine using some kind of look up table).

Maybe we can chat sometime, if you are using element, you can reach out to me via `@fwitte:matrix.org`, or write me an email: `francesco.witte(at)dlr.de`.

**Comment by fwitte** [comment]
@arc12: I want to try out the GitHub discussions features, do you mind if I move this issue?
